file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/requester/AlfrescoFileRequester.java

```diff
@@ -49,21 +52,42 @@
             } else if (responseStatus == NOT_FOUND) {
                 return empty();
             } else {
-                throw new FileOperationException(format("Alfresco is unavailable with response status code: %d", responseStatus.getStatusCode()));
+                throw new FileOperationException(format("Alfresco is unavailable with response status code: %d",
+                        responseStatus.getStatusCode()));
             }
-        } catch (ProcessingException processingException) {
-            throw new FileOperationException("Error fetching resource from Alfresco", processingException);
+        } catch (final ProcessingException processingException) {
+            throw new FileOperationException(format("Error fetching %s from Alfresco with fileId = %s",
+                    fileName, fileId), processingException);
         }
     }
 
-    private String alfrescoUriOf(final String fieldId, final String fileName) {
+    @Override
+    public Optional<InputStream> requestStreamed(String fileId, String fileMimeType, String fileName) {
+        try {
+            final InputStream response = restClient.getAsInputStream(alfrescoUriOf(fileId, fileName, true),
+                    valueOf(fileMimeType), headersWithUserId(alfrescoReadUser));
+            return ofNullable(response);
+        } catch (final NotFoundException nfe) {
+            return empty();
+        } catch (final ProcessingException | InternalServerErrorException ex ) {
+            throw new FileOperationException(format("Error fetching %s from Alfresco with fileId = %s",
+                    fileName, fileId), ex);
+        }
+    }
+
+    private String alfrescoUriOf(final String fieldId, final String fileName, final boolean stream) {
         final StringBuilder requestBuilder = new StringBuilder();
         requestBuilder.append(alfrescoWorkspacePath);
         requestBuilder.append(fieldId);
         requestBuilder.append(URL_SEPARATOR);
         requestBuilder.append("content");
         requestBuilder.append(URL_SEPARATOR);
         requestBuilder.append(fileName);
+        // a means attach -> a = true means don't stream!
+        if (!stream) {
+            requestBuilder.append("?a=true");
```

Comment:
Instead of comment above if, make the "?a=true" string a constant named something like DO_NOT_STREAM